### DISCUSSION TITLE
Disable hibernate envers listeners

### DISCUSSION THREAD

**Discussion by fabrii** [quarkusio/quarkus#43541] [Q&A] [answered]
Hello!

We are trying to disable Envers listeners but keep using the library for retrieving audit data. 

I tried to register a new custom EnversIntegrator, by declaring "org.hibernate.integrator.spi.Integrator" inside META-INF/services. This custom integrator is being invoked, but it is not disabling the listeners.

I guess the only option would be to set "hibernate.envers.autoRegisterListeners" in false? We would need to add this configuration property to Quarkus, right?

Thank you

[a comment by an automated account is not shown]

**Comment by yrodiere** [comment, marked as answer]
Hello,

Do I understand correctly that you want to use Envers in read-only mode, preventing it from adding audit information for changes you make to the database?

That seems dangerous, but if you know what you're doing, then yes, setting `hibernate.envers.autoRegisterListeners` to `true` is the way.

You can try adding this in `application.properties`:

```properties
quarkus.hibernate-orm.unsupported-properties."hibernate.envers.autoRegisterListeners" = false
```

This is, as the name implies, unsupported (untested). If you want this supported in Quarkus, feel free to open a feature request, and ideally a PR.

**Reply by yrodiere** [reply]
I edited my answer, the property needs to be set to `false`, not `true` -- obviously.

**Reply by fabrii** [reply]
Nice! I wasn't aware of `unsupported-properties`. Yes, the idea is to use Envers in read-only mode and insert the audit records through triggers, in order to also audit changes that happen outside the persistence context (e.g., native queries, stored procedures, inserts/updates made by Flyway migrations, etc.).

Thank you very much, as always, for your help.
